lasagne/layers/recurrent.py

```diff
@@ -133,19 +156,27 @@ def get_output_for(self, input, mask=None, *args, **kwargs):
 
         # Create single recurrent computation step function
         def step(layer_input, hid_previous):
-            return self.nonlinearity(
-                self.input_to_hidden.get_output(layer_input) +
-                self.hidden_to_hidden.get_output(hid_previous))
-
-        def step_masked(layer_input, mask, hid_previous):
+            i2h = helper.get_output(
+                self.input_to_hidden, layer_input, **kwargs)
+            h2h = helper.get_output(
+                self.hidden_to_hidden, hid_previous, **kwargs)
+            hid_pre = i2h + h2h
+
+            # clip gradients
+            if self.grad_clipping is not False:
+                hid_pre = theano.gradient.grad_clip(
+                    hid_pre, -self.grad_clipping, self.grad_clipping)
+            return self.nonlinearity(hid_pre)
+
+        def step_masked(layer_input, mask_n, hid_previous):
```

Comment:
I got a warning saying that it masked the outer mask variable. And i figured that input_dot_w was named input_dot_w_n.